src/components/LoginPage/LoginPageComponent.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import slide1 from 'components/LoginPage/images/slide-1.png';
+import slide2 from 'components/LoginPage/images/slide-2.png';
+import slide3 from 'components/LoginPage/images/slide-3.png';
+import slide4 from 'components/LoginPage/images/slide-4.png';
+
+export class LoginPageComponent extends Component {
+  render() {
+    const authHandler = this.props.isGoogleApiInitialized ? this.props.requestAuthorization : null;
```

Comment:
I think this is the place where we should the spinner.

We're disabling an auth handler for the login buttons here.

We could hide the whole component instead and show text "Loading Google API...".

Showing spinner in sagas by applying a side DOM effect doesn't really look cool :)

We have the log of actions and can apply a middleware to show global messages for what is loading (actually we don't really need this messages anyways :).
Calling `showSpinner()` in each saga feels a bit redundant.

Reply:
I agree with all notes except spinner. From my perspective spinner showing us process of saga. So we should use it in all sagas that should show to user loading status. It’s a async business logic and should be in saga, except we don’t need this spinner. 

For better code style and structure, we should use react spinner component. In this case in sagas will be presented only actions to show and hide spinner. Meanwhile, this actions will be saved status to store and react spinner will update his state. 

Am I understand correct?